src/openrct2/localisation/LocalisationService.cpp

```diff
@@ -99,48 +95,59 @@ void LocalisationService::OpenLanguage(int32_t id)
     }
 
     std::string filename;
-    if (id != LANGUAGE_ENGLISH_UK)
-    {
-        filename = GetLanguagePath(LANGUAGE_ENGLISH_UK);
-        _languageFallback = LanguagePackFactory::FromFile(LANGUAGE_ENGLISH_UK, filename.c_str());
-    }
-
     filename = GetLanguagePath(id);
-    _languageCurrent = LanguagePackFactory::FromFile(id, filename.c_str());
-    if (_languageCurrent != nullptr)
+    auto preferredLanguage = LanguagePackFactory::FromFile(id, filename.c_str());
+    if (preferredLanguage != nullptr)
     {
         _currentLanguage = id;
+        _loadedLanguages.emplace_back(std::move(preferredLanguage));
         TryLoadFonts(*this);
     }
     else
     {
         throw std::runtime_error("Unable to open language " + std::to_string(id));
     }
+
+    const auto fallback = LanguagesDescriptors[id].fallback;
+    if (fallback != LANGUAGE_UNDEFINED)
+    {
+        filename = GetLanguagePath(fallback);
+        _loadedLanguages.emplace_back(LanguagePackFactory::FromFile(fallback, filename.c_str()));
+    }
+
+    if (id != LANGUAGE_ENGLISH_UK)
+    {
+        filename = GetLanguagePath(LANGUAGE_ENGLISH_UK);
+        _loadedLanguages.emplace_back(LanguagePackFactory::FromFile(LANGUAGE_ENGLISH_UK, filename.c_str()));
+    }
 }
 
 void LocalisationService::CloseLanguages()
 {
-    _languageFallback = nullptr;
-    _languageCurrent = nullptr;
+    for (auto& language : _loadedLanguages)
+    {
+        language = nullptr;
+    }
+    _loadedLanguages.clear();
     _currentLanguage = LANGUAGE_UNDEFINED;
 }
 
 std::tuple<StringId, StringId, StringId> LocalisationService::GetLocalisedScenarioStrings(
     const std::string& scenarioFilename) const
 {
-    auto result0 = _languageCurrent->GetScenarioOverrideStringId(scenarioFilename.c_str(), 0);
-    auto result1 = _languageCurrent->GetScenarioOverrideStringId(scenarioFilename.c_str(), 1);
-    auto result2 = _languageCurrent->GetScenarioOverrideStringId(scenarioFilename.c_str(), 2);
+    auto result0 = _loadedLanguages[0]->GetScenarioOverrideStringId(scenarioFilename.c_str(), 0);
+    auto result1 = _loadedLanguages[0]->GetScenarioOverrideStringId(scenarioFilename.c_str(), 1);
+    auto result2 = _loadedLanguages[0]->GetScenarioOverrideStringId(scenarioFilename.c_str(), 2);
```

Comment:
Accessing a nonexistent element through `operator[]` is undefined behavior, so we should probably have a check for the vector not being empty before doing this, just like `GetObjectOverrideStringId` does.

Reply:
I don’t think the game even starts up if it cannot load a single language.

Reply:
Ok, but I'd still but a `Guard::Assert` with a proper message.